control-plane/plugin/src/resources/pool.rs

```diff
@@ -179,3 +184,81 @@ pub(crate) fn labels_matched(
     }
     Ok(true)
 }
+
+#[async_trait(?Send)]
+impl Label for Pool {
+    type ID = PoolId;
+    async fn label(
+        id: &Self::ID,
+        label: String,
+        overwrite: bool,
+        output: &utils::OutputFormat,
+    ) -> PluginResult {
+        let result = if label.contains('=') {
+            let [key, value] = label.split('=').collect::<Vec<_>>()[..] else {
+                return Err(TopologyError::LabelMultiAssign {}.into());
+            };
+
+            validate_topology_key(key).context(super::error::PoolLabelFormatSnafu)?;
```

Comment:
note for future refactoring: probably would be useful to put the validation in common code and then apply validation at the backend as well